src/grisp_cryptoauth_cert.erl

```diff
@@ -390,16 +392,18 @@ create_date_vars({generalTime, [Y1,Y2,Y3,Y4,M1,M2,D1,D2,H1,H2,48,48,48,48,90]})
 
 attribute_type_and_value({Key, Value}) ->
     Type = attribute_type(Key),
-    ValueType = case Type of
-                    ?'id-at-dnQualifier'  -> printableString;
-                    ?'id-at-countryName'  -> printableString;
-                    ?'id-at-serialNumber' -> printableString;
-                    _ -> utf8String
+    AttrValue = case Type of
+                    ?'id-at-dnQualifier'    -> Value;  %% printableString
+                    ?'id-at-countryName'    -> Value;  %% printableString
+                    ?'id-at-serialNumber'   -> Value;  %% printableString
+                    ?'id-emailAddress'      -> Value;  %% ia5String
+                    ?'id-domainComponent'   -> Value;  %% ia5String
```

Comment:
Honestly, I wouldn't be surprised, if it works this way. But according to my understanding of the documentation of public_key.hrl this should be tuple `{ia5String, Value}` and `{printableString, Value}`, right?

Reply:
That might not be documented, but it makes sense. Just checkout RFC5280 (which is about the X.509 Certificate structure), for example the definition of the `CommonName`:

```
X520CommonName ::= CHOICE {
      teletexString     TeletexString   (SIZE (1..ub-common-name)),
      printableString   PrintableString (SIZE (1..ub-common-name)),
      universalString   UniversalString (SIZE (1..ub-common-name)),
      utf8String        UTF8String      (SIZE (1..ub-common-name)),
      bmpString         BMPString       (SIZE (1..ub-common-name)) }
```

You have a choice here between different data types and you don't have that choice for other fields like the `CountryName`.